core/components/State.lua

```diff
@@ -0,0 +1,3 @@
+local FSM = require('lib/statemachine')
+
+return FSM
```

Comment:
In practice, the `statemachine` library has been quite significantly modified to accomodate Behaviours. If this is a good thing, these addons should probably exist in this State class, rather than in the original library code.